learnr_bank/learnr_set2.Rmd

```diff
@@ -0,0 +1,468 @@
+```
+
+### Set 2: Answer Key
+
+``` {r, array_q1-answer}
+#Solution to Question 1
+array_numbers <- c(1,2,3,4,5)
+array_numbers
+```
+
+``` {r, array_q2-answer}
+#Solution to Question 2
+array_numbers2 <- 1:5
+array_numbers2
+```
+
+``` {r, array_q3-answer}
+#Solution to Question 3
+even_numbers <- c(2,4,6,8,10)
+even_numbers
+```
+
+``` {r, array_q4-answer}
+#Solution to Question 4
+even_numbers <- seq(2, 10, 2)
+even_numbers
+```
+
+``` {r, array_q5-answer}
+#Solution to Question 5
+mul3 <- seq(0, 999, 3)
+length(mul3)
+```
+
+``` {r, array_q6-answer}
+#Solution to Question 6
+example_array <- c(5, 78, -3, 11, 6, 0)
+example_array / 10
+```
+
+``` {r, array_q7-answer}
+#Solution to Question 7
+example_array <- c(5, 78, -3, 1, 6, 0)
+example_array <= 0
+```
+
+``` {r, array_q8-answer}
+#Solution to Question 8
+example_array <- c(5, 78, -3, 1, 6, 0)
+sum(example_array > 0)
+```
+
+### Set 3: Working with data frames
+
+This set of exercises will use a data frame (aka "tibble" - we will learn the distinction between these terms!) called `msleep`, which is built-in to R. This dataset contains information about some species of mammals as follows:
+
++ `name`: The common name
++ `genus`: The taxonomic genus
++ `vore`: Dietary strategy (carnivore, omnivore or herbivore)
++ `order`: The taxonomic order
++ `conservation`: The species' conservation status
++ `sleep_total`: Total amount of sleep, in hours, the species tends to experience
++ `sleep_rem`: Total amount of REM sleep, in hours, the species tends to experience
++ `sleep_cycle`: Length of sleep cycle, in hours, the species tends to experience
++ `awake`: Amount of time spent awake, in hours, the species tends to experience
++ `brainwt`: Average brain weight, in kilograms, for this species
++ `bodywt`: Average body weight, in kilograms, for this species
+
+
+First, in the code chunk below, examine the data frame using functions we have seen `head()`, `nrow()`, and `ncol()` to get an overall sense of the data frame itself. Then,  answer the following multiple choice questions about this data frame.
+```{r examine-msleep, exercise=T, exercise.lines = 5}
+
+
+
+```
+<!-- the above has no solution - just a chunk for them to practice and answer subsequent multiple choice -->
+
+MC1: How many rows are in this data frame?
+a) 6
+b) 83 [correct]
+c) 11
+d) We can't tell.
+
+MC2: How many columns are in this data frame?
+a) 6
+b) 83 
+c) 11 [correct]
+d) We can't tell.
+
+MC3: How many columns in the data frame contain *numeric* values?
+a) 6 [correct]
+b) 5
+c) All columns are numeric
+d) No columns are numeric
+
+MC4: How many columns in the data frame contain *character* (string) values?
+a) 6 
+b) 5 [correct]
+c) All columns are character
+d) No columns are character
+
+MC5: How many columns in the data frame contain *logical* (TRUE or FALSE) values?
+a) 6 
+b) 5 
+c) All columns are logical
+d) No columns are logical [correct]
+
+```{r quiz1, echo = FALSE}
+quiz(caption = "Multiple Choice",
```

Comment:
Would you like questions 1-5 to be a quiz or individual question chunks? I believe in quizzes I am not able to label the questions with their corresponding number. I might be able to with individual question chunks. I was just asking because the question after the multiple choice (1-5) begins with 6.

Reply:
Let's do individual chunks for these and not a quiz, then.